src/log/client/linux/componolit-gneiss-log-client.adb

```diff
@@ -34,58 +30,27 @@ is
 
    procedure Finalize (C : in out Client_Session)
    is
-      procedure C_Finalize (Label : System.Address) with
-         Import,
-         Convention    => C,
-         External_Name => "finalize",
-         Global        => null;
    begin
-      C_Finalize (C.Label);
-      C.Label  := System.Null_Address;
-      C.Length := 0;
+      if Initialized (C) then
+         Writer.Finalize (C.Session);
+      end if;
    end Finalize;
 
-   procedure Print (Msg : System.Address) with
-      Import,
-      Convention    => C,
-      External_Name => "print",
-      Global        => null;
-
-   procedure Print_With_Null_Term (S : String);
+   procedure Cat (C : in out Client_Session;
```

Comment:
Why `Cat`?

Reply:
The naming refers to `strcat` which concatenates two strings by appending the second string where the first one ends (by checking its null terminator). This behaviour is also employed in `Cat` which looks for the null terminator in the internal buffer and appends the string there.

Reply:
I just find the name confusing as I was thinking of the command line tool `cat`. Maybe `Concatenate` could be cleared?

Reply:
Fixed by https://github.com/Componolit/gneiss/pull/117/commits/2ae856aef468ab1c297ad4f270b3f4fc2bf95d90.